components/tidb_query_vec_expr/src/impl_time.rs

```diff
@@ -148,6 +148,26 @@ pub fn from_days(ctx: &mut EvalContext, arg: Option<&Int>) -> Result<Option<Time
     })
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn month_name(ctx: &mut EvalContext, tmp: Option<&DateTime>) -> Result<Option<Bytes>> {
+    match tmp {
+        Some(t) => {
+            let month = t.month() as usize;
+            if t.is_zero() && ctx.cfg.sql_mode.contains(SqlMode::NO_ZERO_DATE) {
```

Comment:
I don't think we should report errors while we encounter a zero `DateTime` value. At least in MySQL, the query result is `NULL` instead of an error.

Reply:
I can understand that he followed  https://github.com/tikv/tikv/blob/ba2747f7903b393e708b1f10e38f4a47472fcede/components/tidb_query_normal_expr/src/builtin_time.rs#L101-L119
We should unify these in some way.

Reply:
Yes I followed implementation in builtin_time.rs.
I have changed implementation to return `NULL`